src/gui/MainWindow.cc

```diff
@@ -1747,6 +1748,36 @@ void MainWindow::actionShowLibraryFolder()
   QDesktopServices::openUrl(QUrl::fromLocalFile(url));
 }
 
+void MainWindow::actionTreasureTrove()
+{
+  QString dirname (PlatformUtils::backupPath().c_str());
+  QStringList extensions = {};
+  extensions << "scad" << "csg";
+#ifdef ENABLE_PYTHON
+    extensions << "py";
+#endif
+  extensions.replaceInStrings(QRegularExpression("^"), "*.");
+  const auto filter = QString("OpenSCAD Backups (%1)").arg(extensions.join(" "));
+
+  // Bug in QT: dirname is ignored for getOpenFileName
+  QString filename = QFileDialog::getSaveFileName(this, "Open File", dirname, filter);
+
+  if(filename.size() == 0) return; // dont proceed when cancelled
+  
+  tabManager->actionNew();
+
+  std::ostringstream stream; // compile restored design
+  stream << "/*\n"; // put into comments in case it crashes (again)			     
+  std::ifstream file( filename.toStdString());
+  if ( file ) {
+    stream << file.rdbuf();
+    file.close();
+  }
+
+  stream << "\n*/\n";		    
+  activeEditor->setText(stream.str().c_str());
```

Comment:
I think it's easier to keep this within Qt: Use a QString-backed QTextStream. That way you don't have to convert to/from std::string